File: Medium/2026-07-21-3499-Maximize-Active-Section-with-Trade-I/solution.py
class Solution:
    def maximizeActiveSections(self, s: str) -> int:
        # Augment the string with '1's at both ends
        s = '1' + s + '1'
        n = len(s)
        
        # Count the number of active sections
        active_sections = 0
        for i in range(1, n):
            if s[i] == '1' and s[i - 1] == '0':
                active_sections += 1
        
        max_active_sections = active_sections
        
        # Iterate through the string to find blocks of '1's surrounded by '0's
        for i in range(1, n - 1):
            if s[i] == '0' and s[i - 1] == '1' and s[i + 1] == '1':
                # Find the length of the block of '0's
                j = i
                while j < n and s[j] == '0':
                    j += 1
                # Calculate the new number of active sections after the trade
                new_active_sections = active_sections - 1 + 2
                max_active_sections = max(max_active_sections, new_active_sections)
        
        return max_active_sections

File: Medium/2026-07-21-3499-Maximize-Active-Section-with-Trade-I/solution.js
var maximizeActiveSections = function(s) {
    const n = s.length;
    let count = 0;
    
    // Count initial active sections
    for (let i = 0; i < n; i++) {
        if (s[i] === '1' && (i === 0 || s[i - 1] === '0')) {
            count++;
        }
    }
    
    let maxSections = count;
    
    // Augment the string with '1' at both ends
    const augmented = '1' + s + '1';
    const m = augmented.length;
    
    // Iterate through the augmented string
    for (let i = 1; i < m - 1; i++) {
        if (augmented[i] === '0' && augmented[i - 1] === '1' && augmented[i + 1] === '1') {
            let left = i - 1;
            let right = i + 1;
            
            // Expand to the left
            while (left > 0 && augmented[left] === '1') {
                left--;
            }
            // Expand to the right
            while (right < m - 1 && augmented[right] === '1') {
                right++;
            }
            
            // Calculate new sections
            const newSections = (count - 1) + (right - left - 1);
            maxSections = Math.max(maxSections, newSections);
        }
    }
    
    return maxSections;
};